scene-lib/src/scenelib/annotations/el/ATypeElementWithType.java

```diff
@@ -77,10 +77,10 @@ public void setType(Type type) {
     @Override
     public boolean equals(Object o) {
         return o instanceof ATypeElementWithType
-            && ((ATypeElementWithType) o).equals(this);
+            && ((ATypeElementWithType) o).equalsTypeElementWithType(this);
     }
 
-    public boolean equals(ATypeElementWithType o) {
+    public boolean equalsTypeElementWithType(ATypeElementWithType o) {
```

Comment:
With my comment at https://github.com/typetools/annotation-tools/pull/167#discussion_r186309144
I wanted to suggest to inline the separate `equalsTypeElementWithType` method into `equals(Object)`.
What is the point of adding this separate public method if nobody needs it?

Reply:
Hi,
I thought about changing back the name because all the classes in the `AElement` hierarchy use double dispatching mechanism and it looked like a convention, and the double dispatching mechanism is more clear according to Michael.
This class is the only one for which the second equals method isn't used anywhere, and I'm not yet familiar with the complete codebase so I thought it was error prone to change the public api of the class, that's why I changed back the name and made the change so that the `equals(Object)` calls the correct method.
I do have the merged `equals` method in a branch. Should I make a pull request from it?

Reply:
Sure that sounds fine.
Some classes have non-public versions (e.g. `AMethod`and `ATypeElement`) whereas others have public `equalsXXX` methods with javadoc explaining the reason for having a separate method.
As your method is public, how about copying & adapting one of those javadoc comments?